src/topic.ts

```diff
@@ -592,13 +597,27 @@ export class Topic {
     );
   }
 
-  publish(data: Buffer, attributes?: Attributes): Promise<string>;
+  publish(data: Buffer, options?: PubsubMessage): Promise<[string]>;
```

Comment:
I think in this case we should really consider it, especially since we aren't GA yet.

Reply:
If we are going to change the API, then I recommend we get rid of the data field altogether and just have someone send in a PubsubMessage object that we can pass through to publish itself. That way, anything else we add to the message won't require an API change.

Reply:
@kamalaboulhosn re: breaking change

So it would actually pretty easy to avoid a breaking change here with the following overloads

```ts
publish(data: Buffer, attributes?: object)
publish(message: PubsubMessage)
```

The reason I was leaning at breaking at first was because we have a second publish method called [`publishJSON`](https://googleapis.dev/nodejs/pubsub/latest/Topic.html#publishJSON) that already accepts an object as the first param, so it starts to get a little murky there.

Reply:
Is there a typical way that one would do this in node.js? In Python, we could add an optional argument for the ordering key and that would be backwards compatible. If that's an option here and fits with convention, we may want to consider it. Otherwise, the new method that just takes the PubsubMessage seems reasonable to me.

Reply:
I think generally we would either overload or use an object in a similar way to what I'm proposing.

I'm also fine with adding a new method with a deprecation notice on `publish`/`publishJSON`. Any opinions on a name?

@bcoe @stephenplusplus do either of you have any feelings here?

Reply:
In that case, I'm fine going with the new method that just takes the PubsubMessage as you have proposed. We can have the different versions for now and then at some future point deprecate the ones we don't want people to use. I don't think supporting the multiple versions will be too much of a burden for us since they can all ultimately call the one that takes the PubsubMessage.